lib/Barriers/PreDefined.pm

```diff
@@ -113,49 +125,93 @@ The number of the display decimal
 
 The method for the barrier calculation, method_1 or method_2
 
-=head2 config_param
+=head2 contract_barrier_levels
 
-A configuration parameter contains the configuration of each contract type
+A set of barrier level that intended to obtain for a contract type
 
 =cut
 
-my $config_param = LoadFile(File::ShareDir::dist_file('Barriers::PreDefined', 'contract_type_config.yml'));
+has [qw(
+       config_file
+       contract_type
+       duration
+       central_spot
+       display_decimal
+       method
+       )
+     ] => (
+     is  => 'ro'
+    );
+
+has contract_barrier_levels => (
+      is => 'rw',
+      lazy_build => 1,
+);
+
+
+sub _build_contract_barrier_levels {
+    my $self = shift;
+
+    my $config = $self->config_file;
+
+    my $contract_barrier_levels;
+
+    for my $set (@$config) {
+    for my $type (@{ $set->{types} }) {
+        $contract_barrier_levels->{$type} = $set->{levels}
+    }
+}
```

Comment:
It's better to move remove that lazy build. Instead, just have an constructor, which will instantiate & check all what is needed for the module, i.e. 

```perl
has _contract_barrier_levels => ( is => 'rw');

sub BUILD {
    my $self = shift;

    my $config = $self->config_file;

    my $contract_barrier_levels;

    for my $set (@$config) {
        for my $type (@{ $set->{types} }) {
            $contract_barrier_levels->{$type} = $set->{levels}
        }
    }

    $self->_contract_barrier_levels($contract_barrier_levels)
}

```